snapcraft/internal/sources/errors.py

```diff
@@ -93,3 +93,15 @@ class SourceUpdateUnsupportedError(SnapcraftSourceError):
 
     def __init__(self, source):
         super().__init__(source=source)
+
+
+class SnapcraftPullError(SnapcraftSourceError):
+
+    fmt = "Failed to pull source, command {command!r} exited with code {exit_code}."
```

Comment:
this looks inconsistent with other error messages, so you might have just gained a task to search for these inconsistencies :wink: